collector/stat.go

```diff
@@ -98,7 +101,11 @@ func (c *statCollector) Update(ch chan<- prometheus.Metric) (err error) {
 			}
 			// Only some of these may be present, depending on kernel version.
 			cpuFields := []string{"user", "nice", "system", "idle", "iowait", "irq", "softirq", "steal", "guest"}
-			for i, v := range parts[1 : len(cpuFields)+1] {
+			exFieldNum := len(cpuFields)+1
+			if exFieldNum > len(parts) {
```

Comment:
Could you add a comment above this line like:

    // Old kernels lack the "guest" CPU field, which needs to be ignored.

(I'm assuming the "guest" one is the missing one, given that it's the last in the list)